tinyxml_vendor causes nightly clang libcxx build failure

The [nightly clang libcxx build](https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/) is currently failing with the following [output](https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/325/console#console-section-1):

```
--- stderr: tinyxml_vendor
23:57:00 CMake Error at tinyxml-2.6.2-stamp/download-tinyxml-2.6.2.cmake:157 (message):
23:57:00   Each download failed!
23:57:00 
23:57:00     error: downloading 'https://superb-dca2.dl.sourceforge.net/project/tinyxml/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz' failed
23:57:00          status_code: 7
23:57:00          status_string: "Couldn't connect to server"
23:57:00          log:
23:57:00          --- LOG BEGIN ---
23:57:00            Trying 209.61.193.20...
23:57:00 
23:57:00   TCP_NODELAY set
23:57:00 
23:57:00   connect to 209.61.193.20 port 443 failed: Connection timed out
23:57:00 
23:57:00   Failed to connect to superb-dca2.dl.sourceforge.net port 443: Connection
23:57:00   timed out
23:57:00 
23:57:00   Closing connection 0
23:57:00 
23:57:00   
23:57:00 
23:57:00          --- LOG END ---
....
```

I've been able to reproduce this error outside of the buildfarm (Ubuntu 18.04, building ros2 master from source).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ if (NOT TinyXML_FOUND)
   include(ExternalProject)
   ExternalProject_Add(tinyxml-2.6.2
     PREFIX tinyxml-2.6.2
-    URL https://superb-dca2.dl.sourceforge.net/project/tinyxml/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz
+    URL https://downloads.sourceforge.net/project/tinyxml/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz
     URL_MD5 c1b864c96804a10526540c664ade67f0
     CMAKE_ARGS
       -DBUILD_SHARED_LIBS=ON

